### ISSUE TITLE
Salt state output is broken when using onchanges requisite

### ISSUE THREAD

**Issue by genuss** [saltstack/salt#36603]
### Description of Issue/Question

When using ``cmd.run`` with ``onchanges`` and states in the requisite don't provide changes this cmd will not be run. Thus the cmd state will not provide ``start_time`` parameter and the highstate formatter will fail.

### Setup
test.sls
````
echo 1:
  cmd.run:
    - onchanges:
      - file: /tmp/test

/tmp/test:
  file.managed:
    - contents: ''
````

### Steps to Reproduce Issue
run ``salt-call state.apply test`` and in the very end of logs you will find:
````
[DEBUG   ] LazyLoaded highstate.output
[DEBUG   ] Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/output/__init__.py", line 39, in try_printout
    return get_printout(out, opts)(data).rstrip()
  File "/usr/lib/python2.7/dist-packages/salt/output/highstate.py", line 138, in output
    return _format_host(host, hostdata)[0]
  File "/usr/lib/python2.7/dist-packages/salt/output/highstate.py", line 286, in _format_host
    msg = _format_terse(tcolor, comps, ret, colors, tabular)
  File "/usr/lib/python2.7/dist-packages/salt/output/highstate.py", line 555, in _format_terse
    ret)
KeyError: u'start_time'

[DEBUG   ] LazyLoaded nested.output
local:
    ----------
    cmd_|-echo 1_|-echo 1_|-run:
        ----------
        __run_num__:
            1
        __sls__:
            fw
        changes:
            ----------
        comment:
            State was not run because none of the onchanges reqs changed
        result:
            True
    file_|-/tmp/test_|-/tmp/test_|-managed:
        ----------
        __id__:
            /tmp/test
        __run_num__:
            0
        changes:
            ----------
        comment:
            The file /tmp/test is set to be changed
        duration:
            6.598 ms
        name:
            /tmp/test
        pchanges:
            ----------
            newfile:
                /tmp/test
        result:
            None
        start_time:
            15:26:46.241679
````
Notice no ``start_time`` key in ``cmd_|-echo 1_|-echo 1_|-run``.

### Versions Report
minion:
````
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 1.0.0
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian 8.4
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.4
````
master:
````
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 1.0.0
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian 8.5
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.5
````

### Workaround
There is a simple workaround. Use cmd.wait + watch instead of cmd.run.
test.sls
````
echo 1:
  cmd.wait:
    - watch:
      - file: /tmp/test

/tmp/test:
  file.managed:
    - contents: ''
````

**Comment by Ch3LL** [comment]
@genuss you might be running into the same issue as #35989   which has been fixed in #36028  Would you mind giving that fix a try?

**Comment by genuss** [comment]
Thank you @Ch3LL , that's definitely same issue and fix works!
